onnx/defs/generator/defs.cc

```diff
@@ -523,4 +527,185 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, {batch_size, sample_size});
         }));
 
+static const char* Range_ver11_doc = R"DOC(
+Generate a tensor containing a sequence of numbers that begin at `start` and extends by increments of `delta` 
+up to `limit` (exclusive).)DOC";
+
+template <typename T>
+inline int compute_output_dim_for_range(
+    const TensorProto* start,
+    const TensorProto* limit,
+    const TensorProto* delta) {
+  const auto& start_data = ParseData<T>(start);
+  const auto& limit_data = ParseData<T>(limit);
+  const auto& delta_data =
+      delta != nullptr ? ParseData<T>(delta) : std::vector<T>(1, 1);
+
+  if (start_data.size() != 1 || limit_data.size() != 1 ||
+      delta_data.size() != 1) {
+    fail_shape_inference(
+        "Input to 'Range' op should be scalars (Tensor with only one element)");
+  }
+
+  int n = static_cast<int>(
+      ceil((1.0 * (limit_data[0] - start_data[0])) / delta_data[0]));
+  if (n < 0)
+    n = 0;
+
+  return n;
+}
+
+const std::vector<NodeProto> build_nodes_range_op() {
+    // body for 'Loop node'
+    GraphProto loop_sub_graph;
+    loop_sub_graph.set_name("loop body attribute");
+
+    // 'Loop' node 'body' attribute's graph inputs
+    auto* input_value_info_proto_0 = loop_sub_graph.add_input();
+    input_value_info_proto_0->set_name("i");
+
+    auto* input_value_info_proto_1 = loop_sub_graph.add_input();
+    input_value_info_proto_1->set_name("cond");
+
+    auto* input_value_info_proto_2 = loop_sub_graph.add_input();
+    input_value_info_proto_2->set_name("prev");
+
+    // 'Loop' node 'body' attribute's graph nodes
+    auto* node_proto_0 = loop_sub_graph.add_node();
+    node_proto_0->set_op_type("Identity");
+    node_proto_0->add_input();
+    node_proto_0->set_input(0, "cond");
+    node_proto_0->add_output();
+    node_proto_0->set_output(0, "cond_out");
+
+    auto* node_proto_1 = loop_sub_graph.add_node();
```

Comment:
I recommend using the builder convenience functions to keep this compact: e.g., see: https://github.com/onnx/onnx/blob/806aa863020fa180e57f576cb032ec44ce8ddcca/onnx/defs/nn/defs.cc#L2157 or https://github.com/onnx/onnx/blob/b29e78a4efb8e5d8995f576bbf19a846807829b6/onnx/defs/function.h#L69 .
For example, something like:
```
    // current = Add(prev, delta)
    { {"current"}, "Add", {"prev", "delta"}} 
```

Reply:
Will do. Thanks for the suggestion.